src/main/java/command/FindCommand.java

```diff
@@ -0,0 +1,49 @@
+package command;
+
+import java.util.ArrayList;
+
+import exception.DudeException;
+import storage.Storage;
+import storage.TaskList;
+import task.Task;
+import ui.Ui;
+
+/**
+ * Represents a command to find tasks that contain a keyword.
+ */
+public class FindCommand extends Command {
+    private String keyword;
+
+    /**
+     * Creates a FindCommand with the given keyword.
+     *
+     * @param keyword The keyword to search for.
+     */
+    public FindCommand(String keyword) {
+        this.keyword = keyword;
+    }
+
+    /**
+     * Finds tasks that contain the keyword and shows them to the user.
+     *
+     * @param tasks The task list to search for tasks.
+     * @param ui The user interface to interact with the user.
+     * @param storage The storage file to be updated.
+     * @throws DudeException If there is an error finding the tasks.
+     */
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage) throws DudeException {
```

Comment:
Maybe could add more detail to the naming of the execute method? Because every class have a execute method